clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -365,6 +366,11 @@ private void replaceUnresolvedAssignmentWithNewAssignment(
      */
     @Override
     public void transitionToFenced() {
+        if (state == MemberState.PREPARE_LEAVING || state == MemberState.LEAVING) {
+            log.debug("Member ID {} got fenced but it is already leaving the group with state {}," +
```

Comment:
nit: `Member {} with epoch {}`?

Reply:
Sure. Added it also to some other similar messages to make it consistent and provide ID and epoch.